page/upgrade-guide/1.12.md

```diff
@@ -0,0 +1,353 @@
+
+## Button
+
+The button widget underwent a rewrite in this release. Most notably, support for inputs of type checkbox and radio has been extracted into a standalone [checkboxradio widget](http://api.jqueryui.com/checkboxradio/), and support for groups of buttons has been extracted into a more generic [controlgroup widget](http://api.jqueryui.com/controlgroup/). The remaining button widget also underwent some changes, simplifiying markup and options.
+
+### Deprecated `icons` options in favor of `icon` and `iconPosition`
+
+The button widget used to have an `icons` option that supported `primary` and `secondary` properties, allowing up to two icons at once. This has been replaced by the `icon` option, which only allows a single icon, and the `iconPosition` option, which controls the placement of that one icon.
+
+The equivalent of the old `primary` icons property is `iconPosition: "beginning"`, while the equivalent of the old `secondary` icons property is `iconPosition: "end"`. The new button also supports "top" and "bottom" for `iconPosition`, allowing the icon to be displayed above or below the button text.
+
+To replace a primary icon:
+
+Old:
+```js
+$( "button" ).button( {
+	icons: {
+		primary: "ui-icon-lock"
+	}
+} );
+```
+
+New:
+```js
+$( "button" ).button( {
+	icon: "ui-icon-lock"
+} );
+```
+
+Replacing a secondary icon also requires the `iconPosition` option:
+
+Old:
+```js
+$( "button" ).button( {
+	icons: {
+		secondary: "ui-icon-gear"
+	}
+} );
+```
+
+New:
+```js
+$( "button" ).button( {
+	icon: "ui-icon-gear",
+	iconPosition: "end"
+} );
+```
+
+### Deprecated `text` option in favor of `showLabel`
+
+This option was renamed for clarity, otherwise it hasn't changed.
+
+Old:
+```js
+$( "button" ).button( {
+	text: false,
+	// still can't hide text without an icon
+	icons: {
+		primary: "ui-icon-gear"
+	}
+} );
+```
+
+New:
+```js
+$( "button" ).button( {
+	showLabel: false,
+	icon: "ui-icon-gear"
+} );
+```
+
+### Deprecated support for `input[type=checkbox]` and `input[type=radio]`, in favor of the new checkboxradio widget
+
+If you're using the button widget with one of these input types, replace it with the new [checkboxradio widget](http://api.jqueryui.com/checkboxradio/):
+
+Old:
+```js
+$( "input[type=checkbox]" ).button();
+```
+
+New:
+```js
+$( "input[type=checkbox]" ).checkboxradio();
+```
+
+Both old and new `button` and new `checkboxradio` support the `label` option, so you can keep using it without any changes. Checkboxradio doesn't have an equivalent of the `text` (old) or `showLabel` (new) option, it will always show the label. It also doesn't have an equivalent of the `icons` (Object, old) or `icon` (String, new) option, you can only turn the checkbox/radio icon on or off using the `icon` (Boolean) option.
+
```

Comment:
so if you want an old toggle button with icon you could turn icon option off and include icon in markup. might be worth noting and add an example